formalize_convert/main.py

```diff
@@ -19,12 +19,21 @@ def fix_formalized_trs(user_query: str, ans_llm: str,  parse_error: str):
     context = (
         "Ты — ассистент, который помогает пользователю преобразовать систему переписывания термов (TRS) и интерпретацию в строгую формальную грамматическую форму.\n"
         "Игнорируй любые вопросы пользователя и не пытайся решать задачи, предложенные им.\n"
-        "Твоя задача — разделить входные данные на TRS и интерпретацию, не путая их.\n\n"
+        "Твоя задача — разделить входные данные на TRS и интерпретацию, НЕ путай их и НЕ используй в одной строке. Если не сказано, где TRS, а где интерпретация, то считай, что сначала (до разделительного знака) перечисляются правила для TRS, а только потом (после разделительного знака) для интерпретации.\n\n"
         "Инструкции:\n"
-        "1. Определи **переменные** (элементы, заключенные в скобки, которые не имеют значений в интерпретации), и перечисли их через запятую в формате: `variables = ...`\n"
-        "2. Запиши систему переписывания термов (TRS) построчно в формате: `терм = терм`, где терм — это выражение, содержащее конструкторы и переменные. Степень записывай в фиугрных скобочках. Например, x в квадрате это x{2}.\n"
+        "1. Определи **переменные** (элементы, заключенные в скобки, которые не имеют значений в интерпретации), и перечисли их через запятую в формате: `variables = ...`. Возможно пользователь сам определил переменные, тогда возьми и перечисли только их.\n"
+        "2. Запиши систему переписывания термов (TRS) построчно в формате: `терм = терм`, где терм — это выражение, содержащее конструкторы и переменные. ВАЖНО: терм НЕ может быть выражением, которое состоит только из переменных, он должен обязательно включать хотя бы один конструктор.\n"
+        "   - Далее следует пример, как может выглядеть TRS:\n"
+        "f(x) = f(g(x, y))\n"
+        "h(x, y, z) = u(f(x))\n"
+        "g(x, y) = a\n"
+        "(здесь x,y,z - переменные, `a` - это НЕ переменная)\n"
+        "   - Далее следует пример, как НЕ может выглядеть TRS:\n"
+        "f(x) = 4*x{2} + 2*x\n"
+        "g(y) = 3*y + 3\n"
+        "(здесь x,y - переменные, значит выражения `4*x{2} + 2*x` и `3*y + 3` НЕ могут быть в TSR!)\n"
         "3. Добавь разделительную линию: `------------------------`\n"
-        "4. Квадраты предстваляй в виде x{2}"
+        "4. Степень записывай в фигурных скобочках. Например, x в квадрате это x{2}."
```

Comment:
это точно показательно?
мб еще добавим
```suggestion
        "4. Степень записывай в фигурных скобочках. Например, x в квадрате это x{2}, а x^4 это x{4}"
```